src/User/Infra/Security/SecurityUser.php

```diff
@@ -27,11 +27,7 @@ public function __construct(User $user, array $roles = [])
         $credential = $user->getCredential();
         if ($credential instanceof PasswordProtectedInterface) {
             $this->password = $credential->getPassword();
-
-            $algorithm = $credential->getPasswordAlgorithm();
-            if (null !== $algorithm->salt) {
-                $this->passwordSalt = $algorithm->salt->token;
-            }
+            $this->passwordSalt = null === ($salt = $credential->getPasswordAlgorithm()->salt) ? null : $salt->token;
```

Comment:
you can use Null coalescing operator here : 

`$this->passwordSalt = $credential->getPasswordAlgorithm()->salt ?? null;`

Reply:
but we need $salt->token :)

Reply:
oh my fault i meant : 

```php
$this->passwordSalt = $credential->getPasswordAlgorithm()->salt->token ?? null;
```
and yes, it works, even if salt is null : https://3v4l.org/PTRrr

Reply:
done. #216